Source/dom/components/Document.js

```diff
@@ -106,10 +113,92 @@ export class Document extends WrappedObject {
     const wrappedLayer = wrapObject(layer)
     this._object.contentDrawView().centerRect_(wrappedLayer.sketchObject.rect())
   }
+
+  static open(path) {
+    const app = NSDocumentController.sharedDocumentController()
+
+    let document
+
+    if (!path) {
+      app.openDocument()
+
+      document = app.currentDocument()
+    } else {
+      const url = getURLFromPath(path)
+
+      if (app.documentForURL(url)) {
+        return Document.fromNative(app.documentForURL(url))
+      }
+
+      const error = MOPointer.alloc().init()
+
+      document = app.openDocumentWithContentsOfURL_display_error(
+        url,
+        true,
+        error
+      )
+
+      if (error.value() !== null) {
+        throw new Error(error.value())
+      }
+    }
+
+    return Document.fromNative(document)
+  }
+
+  save(path) {
+    let msdocument = this._object
+    const saveMethod =
+      'writeToURL_ofType_forSaveOperation_originalContentsURL_error'
+    if (!msdocument[saveMethod]) {
```

Comment:
That feels like an ugly hack? There's a big difference between `MSDocument` and `MSDocomentData`. If they're conflated in the API into interchangeable object, thats bad!

Reply:
> Yeah, I don’t have much of a choice. When running a plugin I get a MSDocument; when I run the tests, I get a MSDocumentData. I don’t want the dev to bother about the difference